andela/knights-bn-frontend
# #170947596 Configure testing environment

#### What does this PR do?
- This PR sets the testing environment.
#### Description of Task to be completed?
- configure jest framework.
- install test-react-library.
- add test script in the Readme.md.
- test a `Header` component on a landing page example.
- add `Circle CI` continuous integration tool.

#### How should this be manually tested?
- open your favourite terminal
- run `git clone https://github.com/andela/knights-bn-frontend.git` to have the repo on your local machine.
- run `cd knights-bn-frontend` to enter inside the project directory.
- run `git checkout ch-config-tests-environment-170947596` to switch to this branch.
- run `npm install` to install all dependencies required by this branch.
- run `npm start` to start the server and see the landing page example with the centred header to be tested.
- run `npm test` to run the tests.

#### Any background context you want to provide?
N/A

#### Screenshots (if appropriate)
## Landing page example
![image](https://user-images.githubusercontent.com/30593186/79637964-4cf45000-8183-11ea-92da-8aa30df2f284.png)


## Tests result
![image](https://user-images.githubusercontent.com/30593186/79637752-503b0c00-8182-11ea-8712-e6af7f451f1d.png)

#### What are the relevant pivotal tracker stories?
[Test environment story link](https://www.pivotaltracker.com/story/show/170947596)
[Continuous integration story link](https://www.pivotaltracker.com/story/show/170947597)
